src/lib.rs

```diff
@@ -229,6 +231,8 @@ mod lesser;
 mod pread;
 mod pwrite;
 
+#[cfg(feature = "std")]
+pub use crate::buffer::*;
```

Comment:
what's funny in scroll prehistory is it had a module named `buffer` which did something similar (but it was just wrapping `&[u8]` basically, and wasn't dynamic), before I realized I could do everything generically with existing types.